### PR TITLE
Fix Yaml Editor does not respect indent size on new line

### PR DESCRIPTION
**Pull request by lkishalmi**
This one fixes the annoying habit of the Yaml editor that it does not respect the configured indent size.

It seems the `IndentUtils` used the `SPACES_PER_TAB`  in all cases.

Also moved the module language level to Java 17

### CODE REVIEW

File: ide/languages.yaml/src/org/netbeans/modules/languages/yaml/YamlKeystrokeHandler.java [hunk 1]
```diff
@@ -289,9 +290,9 @@ public int beforeBreak(Document document, int offset, JTextComponent target) thr
         int indent = getLineIndent(doc, offset);
         String linePrefix = doc.getText(lineBegin, offset - lineBegin);
         String lineSuffix = doc.getText(offset, lineEnd + 1 - offset);
-        if (linePrefix.trim().endsWith(":") && lineSuffix.trim().length() == 0) {
+        if (linePrefix.stripTrailing().endsWith(":") && lineSuffix.isBlank()) {
```
Comment (mbien) [inline ide/languages.yaml/src/org/netbeans/modules/languages/yaml/YamlKeystrokeHandler.java:293]: gonna steal this and put this into a `stripLeading()`/`stripTrailing()` rule -> https://github.com/mbien/jackpot-inspections/commit/037b5867190b49030be754df8c18c9a6646df250 :)

**Review by mbien — APPROVED** [review]
looks good!

**Review by matthiasblaesing — APPROVED** [review]
Looks sane to me.

As a side comment: `Utilities#getRowStart` in `InsertTabAction` is deprecated and can be trivially replaced with `LineDocumentUtils#getLineStart`.

**Comment by mbien** [comment]
well, this PR had no milestone and looks semi-ready. What shall we do now, merge or poke @lkishalmi to fix the deprecation?

**Comment by matthiasblaesing** [comment]
As I brought up the deprecation fix, I don't think this should hold up merging. It was just a side observation.

**Comment by lkishalmi** [comment]
Fixed some deprecations as well.

**Pull request merged**